src/main/scala/com/databricks/labs/overwatch/validation/DeploymentValidation.scala

```diff
@@ -0,0 +1,528 @@
+    val workspaceId = row.getAs(ConfigColumns.workspace_id.toString).toString
+    val testDetails = s"""WorkSpaceURLConnectivityTest APIURL:${url} DBPATWorkspaceScope:${scope} SecretKey_DBPAT:${patKey}"""
+    try {
+      val patToken = dbutils.secrets.get(scope = scope, key = patKey)
+      val apiEnv = ApiEnv(false, url, patToken, getClass.getPackage.getImplementationVersion)
+      val endPoint = "clusters/list"
+      ApiCallV2(apiEnv, endPoint).execute().asDF()
+      DeploymentValidationReport(true,
+        getSimpleMsg("APIURL_Connectivity"),
+        testDetails,
+        Some("SUCCESS"),
+        Some(workspaceId)
+      )
+    } catch {
+      case exception: Exception =>
+        val msg = s"""No Data retrieved workspaceId:${workspaceId} APIURL:${url} DBPATWorkspaceScope:${scope} SecretKey_DBPAT:${patKey}"""
+        val fullMsg = PipelineFunctions.appendStackStrace(exception, msg)
+        logger.log(Level.ERROR, fullMsg)
+        DeploymentValidationReport(false,
+          getSimpleMsg("APIURL_Connectivity"),
+          testDetails,
+          Some(fullMsg),
+          Some(workspaceId)
+        )
+
+    }
+
+  }
+
+  /**
+   * Checks for distinct values in the provided column.
+   * @param ruleName
+   * @param configColumns
+   * @return
+   */
+  private def validateDistinct(ruleName: String, configColumns: String): Rule = {
+    Rule(ruleName, countDistinct(configColumns), lit(1))
+  }
+
+
+  /**
+   * Cloud should be either AWS or Azure.
+   * @return
+   */
+  private def validateCloud(): Rule = {
+    Rule("Valid_Cloud_providers", lower(col(ConfigColumns.cloud.toString)), Array("aws", "azure"))
+  }
+
+
+  /**
+   * PrimordialDate should be less than current date and should be in yyyy-MM-dd format.
+   * @return
+   */
+  private def validatePrimordialDate(): Rule = {
+    Rule("Valid_PrimordialDate", to_date(col(ConfigColumns.primordial_date.toString), "yyyy-MM-dd") <= current_date() && to_date(col(ConfigColumns.primordial_date.toString), "yyyy-MM-dd").isNotNull)
+  }
+
+  /**
+   * Check for not null values in the provided column
+   * @param ruleName
+   * @param configColumns
+   * @return
+   */
+  private def validateNotNull(ruleName: String, configColumns: String): Rule = {
+    Rule(ruleName, col(configColumns).isNotNull)
+  }
+
+  /**
+   *Max days should be greater then 0 and should be not null.
+   * @return
+   */
+  private def validateMaxDays(): Rule = {
+    Rule("Valid_MaxDays", col(ConfigColumns.max_days.toString) >= 0 && col(ConfigColumns.max_days.toString).isNotNull)
+  }
+
+  /**
+   * Scopes for Overwatch should be audit,sparkEvents,jobs,clusters,clusterEvents,notebooks,pools,accounts,dbsql
+   * @return
+   */
+  private def validateOWScope(): Rule = {
+    Rule("Valid_Excluded_Scopes", lower(col(ConfigColumns.excluded_scopes.toString)), Array("audit", "sparkEvents", "jobs", "clusters", "clusterEvents", "notebooks", "pools", "accounts", "dbsql","",null))
```

Comment:
line is way too long, please add reasonable line breaks.

Reply:
done